### PR TITLE
Add support for react 18

### PR DESCRIPTION
**Pull request by coolusaHD**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,40 +49,40 @@
...
+    "ts-jest": "^28.0.5",
+    "ts-loader": "^9.3.1",
+    "ts-node": "^10.9.1",
+    "typescript": "^4.7.4",
+    "vite": "^3.0.0"
   },
   "peerDependencies": {
-    "react": "^16.14.0 || ^17.0.0",
-    "react-dom": "^16.14.0 || ^17.0.0"
+    "react": "^18.0.0",
```
Comment (todd-elvers) [inline package.json]: Will this emit an npm warning during install on projects that aren't on React 18 yet?  Could this be `^17 || ^18` in the `peerDependencies` but left as `^18.0.0` in the `devDependencies`?

Comment (todd-elvers) [reply package.json]: Ideally we could support React 17 _and_ 18 compatible

**Review by todd-elvers — COMMENTED** [review]
Great work!  I added a single comment about the React version in `peerDependencies` but otherwise this looks great

File: package.json (hunk 1 above)
Comment (coolusaHD) [reply package.json]: I think this should work with adding `^17 || ^18`
I will test it and paste the resut here.

**Comment by coolusaHD** [comment]
should now be workig on both react versions
please recheck before merge @todd-elvers 
🚀

**Review by todd-elvers — COMMENTED** [review]
LGTM

**Pull request merged**